executor/grant.go

```diff
@@ -132,15 +134,15 @@ func (e *GrantExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	}
 	defer func() {
 		if !isCommit {
-			_, err := internalSession.(sqlexec.SQLExecutor).ExecuteInternal(context.Background(), "rollback")
+			_, err := internalSession.(sqlexec.SQLExecutor).ExecuteInternal(ctx, "rollback")
```

Comment:
I'm not sure if it's ok to change these background ctx to the parent ctx... Could the original intention be just allowing it even if the parent ctx is done?

Reply:
Emmm, it seems many `RunInNewTxn` and `ExecuteInternal` use context in a messy way(if there is no context in the parameters, create a new one), creating a new context in many places is redundant. Anyway, all the modifies have been checked for compatibility.